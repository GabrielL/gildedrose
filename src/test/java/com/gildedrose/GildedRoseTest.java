package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import java.nio.charset.StandardCharsets;

class GildedRoseTest {

    @Test
    void foo() {
        Item[] items = new Item[] { new Item("foo", 0, 0) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        Approvals.verifyAll("items", app.items);
    }

    @Test
    void approvalOriginalMain() {
        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
        PrintStream out = new PrintStream(outStream, true, StandardCharsets.UTF_8);
        GildedRoseMain.gildedRoseMain(out, 2);
        Approvals.verify(outStream);
    }

    @Test
    void matchLine26() {
        Item[] items = new Item[]{new Item("Backstage passes to a TAFKAL80ETC concert", 0, 0)};
        
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        Approvals.verifyAll("items", app.items);
    }

}
