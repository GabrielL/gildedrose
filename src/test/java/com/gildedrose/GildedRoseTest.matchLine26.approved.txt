items[0] = Backstage passes to a TAFKAL80ETC concert, -1, 0
